### ISSUE TITLE
extra is nil，runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by zituocn** [qiniu/go-sdk#81]
storage/form_upload.go

line:398

*原始代码*：

```go
      //extra.Params
	if extra.Params != nil {
		for k, v := range extra.Params {
			if (strings.HasPrefix(k, "x:") || strings.HasPrefix(k, "x-qn-meta-")) && v != "" {
				err = writer.WriteField(k, v)
				if err != nil {
					return
				}
			}
		}
	}
```

*修正后的*：

```go
      //extra.Params
	if extra!=nil&&extra.Params != nil {
		for k, v := range extra.Params {
			if (strings.HasPrefix(k, "x:") || strings.HasPrefix(k, "x-qn-meta-")) && v != "" {
				err = writer.WriteField(k, v)
				if err != nil {
					return
				}
			}
		}
	}

```

**Comment by YangSen-qn** [comment]
感谢您的反馈

**Comment by bachue** [comment]
fixed in v7.15.0

**Comment by zituocn** [comment]
> fixed in v7.15.0

NICE
